pkg/observer.go

```diff
@@ -0,0 +1,83 @@
+package ocr2keepers
+
+import "context"
+
+type UpkeepPayload interface{}
+type CheckResult interface{}
+
+type Tick interface {
+	// GetUpkeeps provides an array of upkeeps scoped to the individual tick
+	GetUpkeeps(ctx context.Context) ([]UpkeepPayload, error)
+}
+
+// Runner2 is the interface for an object that should determine eligibility state
+// (I didn't want to mess with the existing runner and was unsure if we wanted to reuse that)
+type Runner2 interface {
+	// CheckUpkeeps has an input of upkeeps with unknown state and an output of upkeeps with known state
+	CheckUpkeeps(context.Context, []UpkeepPayload) ([]CheckResult, error)
+}
+
+// Preprocessor is the general interface for middleware used to filter, add, or modify upkeep
+// payloads before checking their eligibility status
+type Preprocessor interface {
+	// PreProcess takes a slice of payloads and returns a new slice
+	PreProcess(context.Context, []UpkeepPayload) ([]UpkeepPayload, error)
+}
+
+// Postprocessor is the general interface for a processing function after checking eligibility status
+type Postprocessor interface {
+	// PostProcess takes a slice of results where eligibility status is known
+	PostProcess(context.Context, []CheckResult) error
+}
+
+type Observer interface {
```

Comment:
This isn't needed in this file since the observer interface isn't a dependency of the observer.